src/app/wizard/deploy/deploy.component.ts

```diff
@@ -11,52 +12,64 @@ import {KeycloakService} from "../../shared/keycloak.service";
 export class DeployComponent {
   @Input() submittedGuis: Gui[];
   @Input() command: string;
-  consoleUrl: string;
-  statusList: Status[] = [
-    new Status("Generating source"),
-    new Status("Create github repo"),
-    new Status("Pushing the generated code"),
-    new Status("Creating openshift project")
-  ];
+  progress: boolean;
+  statusMessages: StatusMessage[];
+
+  private apiUrl: string = process.env.LAUNCHPAD_MISSION_CONTROL_URL;
+  private webSocket: WebSocket;
 
   constructor(private forgeService: ForgeService,
-              private kc: KeycloakService) {
+    private kc: KeycloakService,
+    private config: Config) {
+      if (!this.apiUrl) {
+        this.apiUrl = config.get('mission_control_url');
+      }
+      if (this.apiUrl && this.apiUrl[this.apiUrl.length - 1] != '/') {
+        this.apiUrl += '/';
+      }
   }
 
   deploy(): void {
-    this.forgeService.upload(this.command, this.submittedGuis)
-    .then(url => {
-      window.location.href = url;
-    });
-  }
-
-  downloadZip(): void {
-    this.forgeService.downloadZip(this.command, this.submittedGuis);
-  }
-
-  login() {
-    this.kc.login();
-  }
-
-  isAuthenticated():boolean {
-    return this.kc.isAuthenticated();
+    if (this.kc.isAuthenticated()) {
+      this.progress = true;
+      this.forgeService.upload(this.command, this.submittedGuis)
+        .then(status => {
+          this.webSocket = new WebSocket(this.apiUrl + status.uuid_link);
```

Comment:
@quintesse yes connect to `ws://localhost:8080/status/<guid>` the `StatusResult` contains that url part. So if we have reverse proxy the `LAUNCHPAD_MISSIONCONTROL_URL` can be empty